code_samples/api/rest_api/src/Rest/InputParser/GreetingInput.php

```diff
@@ -9,7 +9,7 @@
 
 class GreetingInput extends BaseParser
 {
-    public function parse(array $data, ParsingDispatcher $parsingDispatcher)
+    public function parse(array $data, ParsingDispatcher $parsingDispatcher): mixed
```

Comment:
```suggestion
    public function parse(array $data, ParsingDispatcher $parsingDispatcher): Greeting
```
Technicaly, you can do it like that to improve on future typehinting. As we are using `mixed` in interface, it is completly safe to narrow type.

`mixed` is fine also, so it is up to you.